core/edit/edit_command_actions.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import dataclass
+from typing import Callable, Union
+
+from talon import Module, actions
+
+
+@dataclass
+class EditSimpleAction:
+    type: str
+
+    def __str__(self):
+        return self.type
+
+
+@dataclass
+class EditInsertAction:
+    type = "insert"
+    text: str
+
+    def __str__(self):
+        return self.type
+
+
+@dataclass
+class EditFormatAction:
+    type = "applyFormatter"
+    formatters: str
+
+    def __str__(self):
+        return self.type
+
+
+EditAction = Union[
+    EditSimpleAction,
+    EditInsertAction,
+    EditFormatAction,
+]
```

Comment:
```suggestion
@dataclass
class EditAction:
    type: str

    def __str__(self):
        return self.type


@dataclass
class EditInsertAction(EditAction):
    type = "insert"
    text: str


@dataclass
class EditFormatAction(EditAction):
    type = "applyFormatter"
    formatters: str
```

Reply:
Why? `EditAction` is was get pasts into the action and needs to be of every kind.

Reply:
If they subclass `EditAction` then `EditAction` is effectively already a union of every kind.

Reply:
True. I missed the inheritance part. I basically only do Talon in python and there isn't much call for inheritance there :D
You definitely right that that would work, but I'm not sure if it's strictly better.